src/app/code/community/Divante/VueStorefrontIndexer/Model/Index/Mapping/Review.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+use Divante_VueStorefrontIndexer_Api_MappingInterface as MappingInterface;
+use Divante_VueStorefrontIndexer_Api_Mapping_FieldInterface as FieldInterface;
+use Divante_VueStorefrontIndexer_Model_Index_Mapping_Eav_Abstract as AbstractMapping;
+use Divante_VueStorefrontIndexer_Model_Index_Mapping_Generalmapping as GeneralMapping;
+
+/**
+ * Class Divante_VueStorefrontIndexer_Model_Index_Mapping_Product
+ *
+ * @package     Divante
+ * @category    VueStoreFrontIndexer
+ * @author      Jose Castaneda <jose@qbo.tech>
+ * @copyright   Copyright (C) 2018 Divante Sp. z o.o.
+ * @license     See LICENSE_DIVANTE.txt for license details.
+ */
+class Divante_VueStorefrontIndexer_Model_Index_Mapping_Review implements MappingInterface
+{
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var array
+     */
+    private $properties;
+
+    /**
+     * @inheritdoc
+     */
+    public function setType($type)
+    {
+        $this->type = $type;
+    }
+
+    /**
+     * @return string
+     */
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getMappingProperties()
+    {
+        if (null === $this->properties) {
+            $attributes = $this->getAttributes();
+            $attributesMapping  = [];
+
+            foreach ($attributes as $attributeCode => $attribute) {
+                $attributesMapping = array_merge($attributesMapping, $this->getAttributeMapping($attributeCode, $attribute));
+            }
```

Comment:
We don't need to use array_merge here
we can just do:
```
 foreach ($attributes as $attributeCode => $attribute) {
                $attributesMapping[attributeCode] =  $this->getAttributeMapping($attributeCode, $attribute);
            }
```

and `getAttributeMapping` can only return the properties like:
```
 public function getAttributeMapping($attributeCode, $attribute)
    {
        $mapping = [];
        $type = $attribute['type'];
        if ($type === 'text') {
            $fieldName = $attributeCode;
            $mapping = [
                'type' => $type,
                'fielddata' => true,
                'fields' => [
                    'keyword' => [
                        'type' => FieldInterface::TYPE_KEYWORD,
                        'ignore_above' => 256
                    ]
                ]
            ];
        
        } else {
            $mapping = ['type' => $type];
        }
        return $mapping;
    }
```

OR we can skip `attributeMapping` function  and we can do sth like this:
```
       $properties = [
            'created_at' => [
                'type' => FieldInterface::TYPE_DATE,
                'format' => FieldInterface::DATE_FORMAT,
            ],
            'id' => ['type' => FieldInterface::TYPE_LONG],
            'product_id' => ['type' => FieldInterface::TYPE_LONG],
            'title' => ['type' => FieldInterface::TYPE_TEXT],
            'detail' => ['type' => FieldInterface::TYPE_TEXT],
            'nickname' => ['type' => FieldInterface::TYPE_TEXT],
            'review_entity' => ['type' => FieldInterface::TYPE_KEYWORD],
            'review_status' => ['type' => FieldInterface::TYPE_INT],
            'customer_id' => ['type' => FieldInterface::TYPE_INT],
        ];
```